client/src/client.rs

```diff
@@ -1,54 +1,57 @@
-use crate::errors::Result;
+use base64::{prelude::BASE64_STANDARD, Engine as _};
+use core::str;
+use libsignal_core::{Aci, Pni, ProtocolAddress, ServiceId};
+use rand::{rngs::OsRng, Rng};
+use sqlx::{migrate::MigrateDatabase, Sqlite, SqlitePool};
+use std::collections::HashMap;
+
 use crate::{
     contact_manager::ContactManager,
     encryption::{encrypt, pad_message, unpad_message},
-    errors::{ProcessPreKeyBundleError, ReceiveMessageError},
+    errors::{ProcessPreKeyBundleError, ReceiveMessageError, Result, SignalClientError},
     key_manager::KeyManager,
     server::{SignalServer, SignalServerAPI},
     storage::{
-        generic::{ProtocolStore, Storage, StorageType},
-        in_memory::InMemory,
+        database::ClientDB,
+        device::Device,
+        generic::{ProtocolStore, Storage},
     },
 };
-use base64::{prelude::BASE64_STANDARD, Engine as _};
 use common::{
     signalservice::{envelope, Content, DataMessage},
     web_api::{
         AccountAttributes, DeviceCapabilities, RegistrationRequest, SignalMessage, SignalMessages,
     },
 };
-use core::str;
-use libsignal_core::{Aci, Pni, ProtocolAddress, ServiceId};
 use libsignal_protocol::{
     message_decrypt, process_prekey_bundle, CiphertextMessage, CiphertextMessageType,
     IdentityKeyPair, SignalProtocolError,
 };
 use prost::Message;
-use rand::{rngs::OsRng, Rng};
 use std::time::{SystemTime, UNIX_EPOCH};
 
-pub struct Client<ST: StorageType, SSA: SignalServerAPI> {
+pub struct Client<T: ClientDB, SSA: SignalServerAPI> {
```

Comment:
```suggestion
pub struct Client<T: ClientDB, U: SignalServerAPI> {
```
og husk andre steder

Reply:
Done